### PR TITLE
feat(e2e/admin): add upgrade portal action

### PR DESCRIPTION
**Pull request by kevinhalliday**
Add upgrade portal action. 

Additional:

- refactor to reduce code duplication and add tests

issue: #1393

### CODE REVIEW

File: contracts/core/script/admin/Admin.s.sol [hunk 1]
```diff
@@ -11,21 +15,47 @@ import { IOmniPortalAdmin } from "src/interfaces/IOmniPortalAdmin.sol";
...
-     * @param portal The address of the portal contract.
+     * @param admin     The owner of the portal contract.
+     * @param portal    The address of the portal contract.
      */
-    function unpausePortal(address portal) public {
-        vm.startBroadcast();
-        IOmniPortalAdmin(portal).pause();
+    function unpausePortal(address admin, address portal) public {
+        vm.startBroadcast(admin);
+        OmniPortal(portal).unpause();
```
Comment (ttarsi) [inline contracts/core/script/admin/Admin.s.sol:34]: 👀

File: contracts/core/script/admin/Admin.s.sol [hunk 2]
```diff
@@ -11,21 +15,47 @@ import { IOmniPortalAdmin } from "src/interfaces/IOmniPortalAdmin.sol";
...
+    function upgradePortal(address admin, address deployer, address portal, bytes calldata data) public {
+        // deploy new implementation
+        vm.startBroadcast(deployer);
+        address impl = address(new OmniPortal());
+        vm.stopBroadcast();
+
+        // upgrade proxy
+        vm.startBroadcast(admin);
+        address proxyAdmin = EIP1967Helper.getAdmin(portal);
+        ProxyAdmin(proxyAdmin).upgradeAndCall(ITransparentUpgradeableProxy(portal), impl, data);
```
Comment (ttarsi) [inline contracts/core/script/admin/Admin.s.sol:54]: would we ever use upgrade and call? or is that only used to initialize

**Review by ttarsi — APPROVED** [review]

File: e2e/fbproxy/app/proxy.go [hunk 3]
```diff
@@ -58,11 +58,29 @@ func (p *proxy) Proxy(w http.ResponseWriter, r *http.Request) {
...
+	for {
+		if err == nil {
+			return false
+		}
+		if errors.Is(err, context.Canceled) {
+			return true
+		}
+		err = errors.Unwrap(err)
+	}
+}
```
Comment (corverroos) [inline e2e/fbproxy/app/proxy.go:82]: `errors.Is(err, context.Canceled)` should be sufficient, it does check the whole chain, no need to `Unwrap` yourself.

**Review by corverroos — APPROVED** [review]

File: contracts/core/script/admin/Admin.s.sol (hunk 1 above)
Comment (kevinhalliday) [reply contracts/core/script/admin/Admin.s.sol:34]: gonna add some tests for these

**Pull request merged**